### PR TITLE
Refactor upwinding test

### PR DESCRIPTION
**Pull request by OmarDuran**
## Proposed changes

An effort was made to simplify and compress the tests. The following file was refactored:
- `test_upwind.py`

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [ ] The documentation is up-to-date.
- [ ] Static typing is included in the update.
- [ ] This PR does not duplicate existing functionality.
- [ ] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

File: tests/numerics/fv/test_upwind.py [hunk 1]
```diff
@@ -0,0 +1,682 @@
...
+    def _compose_algebraic_representation(self, mdg, upwind_obj, upwind_coupling_obj):
+        """Create discretization matrix and rhs vector from mdg, upwind and
+            upwind_coupling objects.
+
+        This method essentially mimics the functionality of the antiquated assembler class
+        """
+
+        sd_hashes = [hash(sd) for sd in mdg.subdomains()]
+        intf_hashes = [hash(intf) for intf in mdg.interfaces()]
+        hashes = sd_hashes + intf_hashes
```
Comment (keileg) [inline tests/numerics/fv/test_upwind.py:488]: Do we need to use hashes here? Why not just lists?

Comment (keileg) [reply tests/numerics/fv/test_upwind.py:488]: I think I see why (below). Please leave a short documentation (here and below).

**Review by keileg — APPROVED** [review]
Please comment on the use of hashes. Besides, look good.

File: tests/numerics/fv/test_upwind.py (hunk 1 above)
Comment (OmarDuran) [reply tests/numerics/fv/test_upwind.py:491]: Thanks for the comment, I have improved the documentation.

**Pull request merged**